cli/tests/test_git_fetch.rs

```diff
@@ -51,6 +52,33 @@ fn add_git_remote(test_env: &TestEnvironment, repo_path: &Path, remote: &str) {
     );
 }
 
+/// Clone a git repo from a source, add a commit on top
+fn clone_git_add_commit(test_env: &TestEnvironment, src_remote: &str, remote: &str) {
+    let src_repo_path = test_env.env_root().join(src_remote);
+    let git_repo_path = test_env.env_root().join(remote);
+    let git_repo = git2::Repository::clone(src_repo_path.to_str().unwrap(), git_repo_path).unwrap();
+    let signature =
+        git2::Signature::new("Some One", "some.one@example.com", &git2::Time::new(0, 0)).unwrap();
+    let mut tree_builder = git_repo.treebuilder(None).unwrap();
+    let file_oid = git_repo.blob(remote.as_bytes()).unwrap();
+    tree_builder
+        .insert("file", file_oid, git2::FileMode::Blob.into())
+        .unwrap();
+    let tree_oid = tree_builder.write().unwrap();
+    let tree = git_repo.find_tree(tree_oid).unwrap();
+    // our branch name is the same as the source branch name
+    git_repo
+        .commit(
+            Some(&format!("refs/heads/{src_remote}")),
+            &signature,
+            &signature,
+            "message",
+            &tree,
+            &[],
+        )
+        .unwrap();
```

Comment:
nit: probably better to extract `add_git_commit()` function. I don't think `clone` + `add_commit` is common operation.